### PR TITLE
Updating the out_oms packages and Golang version to 1.21

### PR DESCRIPTION
**Pull request by jatakiajanvi12**
This pull request includes significant changes to the `go.mod` file and `utils.go` file in the `source/plugins/go/src` directory of the project. The main changes involve updating the Go version, updating dependencies, and modifying the way the ADX client is created.

Here are the most important changes:

Go version and dependencies update:

* [`source/plugins/go/src/go.mod`](diffhunk://#diff-4c8eb2bbf5d4a6f8f0d84217e96f7fb96f5d919c8f8e2519e5f99e86cd3f9262L3-R78): The Go version has been updated from 1.14 to 1.21. Several dependencies have been updated, including `github.com/Azure/azure-kusto-go`, `github.com/fluent/fluent-bit-go`, and `github.com/google/uuid`. Some dependencies have been removed, while new ones like `k8s.io/api`, `k8s.io/apimachinery`, and `k8s.io/client-go` have been added.

Changes in the ADX client creation:

* [`source/plugins/go/src/utils.go`](diffhunk://#diff-de3771438420bafe6137baebb8599dd304fe27cc69d837d0a9cbe11bcbbf6abdL20): The import statement for `github.com/Azure/go-autorest/autorest/azure/auth` has been removed. The `CreateADXClient` function has been modified to use `kusto.NewConnectionStringBuilder` for creating the ADX client instead of `auth.NewClientCredentialsConfig`. [[1]](diffhunk://#diff-de3771438420bafe6137baebb8599dd304fe27cc69d837d0a9cbe11bcbbf6abdL20) [[2]](diffhunk://#diff-de3771438420bafe6137baebb8599dd304fe27cc69d837d0a9cbe11bcbbf6abdL196-R197)

These changes seem to be aimed at updating the project to use newer versions of Go and its dependencies, and at changing the way the ADX client is created for better compatibility with the updated dependencies.

### CODE REVIEW

**Comment by pfrcks** [comment]
can you run: go mod tidy?

File: source/plugins/go/src/go.mod [hunk 1]
```diff
@@ -4,30 +4,366 @@ go 1.14
 
 require (
 	github.com/Azure/azure-kusto-go v0.5.2
+	github.com/Azure/go-autorest/autorest/azure/auth v0.5.12
+	github.com/Microsoft/go-winio v0.5.2
+	github.com/fluent/fluent-bit-go v0.0.0-20171103221316-c4a158a6e3a7
+	github.com/golang/mock v1.6.0
+	github.com/google/uuid v1.4.0
+	github.com/microsoft/ApplicationInsights-Go v0.4.4
+	github.com/tinylib/msgp v1.1.9
```
Comment (ganga1980) [inline source/plugins/go/src/go.mod:12]: 1.  Does this update addresses the reported CVEs? 2. can you please do test for container log and prom metrics?

**Comment by jatakiajanvi12** [comment]
> can you run: go mod tidy?

tried running this but since I have a higher version of go than 1.14 it updates the mod file to add toolchain, etc. but with that change the build is failing. So, decided not to take that

File: source/plugins/go/src/go.mod (hunk 1 above)
Comment (jatakiajanvi12) [reply source/plugins/go/src/go.mod:12]: I think so because this is not caught by trivy....Have tested the log and metrics scenario

**Review by ganga1980 — APPROVED** [review]

**Pull request merged**